tests/PregTest.php

```diff
@@ -50,7 +50,7 @@ public function testMatchNoMatch()
     public function testMatchThrowsWithBadPatternIfWarningsAreNotThrowing()
     {
         $this->doExpectException('Composer\Pcre\PcreException', 'preg_match(): failed executing "{abc": '.(PHP_VERSION_ID >= 80000 ? 'Internal error' : (PHP_VERSION_ID >= 70201 ? 'PREG_INTERNAL_ERROR' : '' /* Ignoring here, some old versions return UNDEFINED_ERROR while some have been fixed */)));
-        @Preg::match('{abc', 'abcdefghijklmnopqrstuvwxyz');
+        @Preg::match('{abc', 'abcdefghijklmnopqrstuvwxyz', $matches);
```

Comment:
Here I wouldn't add $matches either, or is there a reason you added it?

Reply:
Yes, Preg::match requires at least 3 arguments, otherwise

```1) Composer\Pcre\PregTest::testMatchThrowsWithBadPatternIfWarningsAreNotThrowing
Failed asserting that exception of type "ArgumentCountError" matches expected exception "Composer\Pcre\PcreException". Message was: "Too few arguments to function Composer\Pcre\Preg::match(), 2 passed in C:\Users\John\projects\pcre\tests\PregTest.php on line 53 and at least 3 expected" at
C:\Users\John\projects\pcre\src\Preg.php:26
C:\Users\John\projects\pcre\tests\PregTest.php:53
```

Reply:
I think this fails because you made $matches non nullable in `Preg::match`, but https://www.php.net/preg_match has `$matches` as `&$matches=null` so that it is optional.

Reply:
Yes, I've got confused somewhere along the way! I'll make the by-refs nullable